src/index.js

```diff
@@ -152,6 +152,20 @@ class Package {
 	this.payload			= payload
     }
 
+    metadata ( key, value ) {
+	if ( arguments.length === 2 ) {
+	    if ( value === undefined ) {
+		const previous_value	= this._metadata[key];
+		delete this._metadata[key];
+		return previous_value;
+	    }
```

Comment:
Can you put this into context for me? When might the metadata return without a value, and why how do we know it's `response_id` will be/should be the exact same?

Reply:
I don't fully understand the question.  The metadata value is always returned, but the value may have been set to `undefined`